fs/graph.go

```diff
@@ -269,6 +269,40 @@ func (g *Graph) AddReferencingTests() []*File {
 	return tests
 }
 
+// addEphemeralUpstreamsFor recursively adds upstream ephemeral nodes to a given node
+func (g *Graph) addEphemeralUpstreamsFor(node *Node) error {
```

Comment:
Do you need the `error` return here, it appears we're always going to return nil no matter what, which means we can remove the error from `AddEphemeralUpstreams` too